### DISCUSSION TITLE
How do you get Gtk4LayerShell to work with java-gtk?

### DISCUSSION THREAD

**Discussion by foxyfocus** [bailuk/java-gtk#41] [Q&A]
So, I was able to generate bindings for Gtk4LayerShell using the .gir file found here: [https://github.com/pentamassiv/gtk4-layer-shell-gir](url) However, I am unable to use the bindings due to the following error: 
![gtklayershell](https://github.com/user-attachments/assets/451ae075-b9c1-426b-b73d-3ca1c280c9ad)

If anyone is able to try and get Gtk4LayerShell working, that would be greatly appreciated.

**Comment by foxyfocus** [comment]
Update: 

So, I think I may have found the issue: 
![javalink](https://github.com/user-attachments/assets/5e7c3dc8-660f-4f67-9c3d-efe78d2345f2)
The library name should be gtk4-layer-shell, not gtk4-layer-shell-0. However, I am not sure what I need to change when generating the bindings and building. Do I need to change the .gir file directly to reflect this change when java-gtk is built? Or would I be able to change it via the generator/src/main/kotlin/ch/bailu/gtk/Configuration.kt source file? I will try both and report back.

**Comment by foxyfocus** [comment]
Okay, so I figured out the issue and was able to fix it by changing the generator/src/main/kotlin/ch/bailu/gtk/Configuration.kt.

**Comment by bailuk** [comment]
Yes the most common name should be in Configuration.kt

Well known alternative names can also be added to https://github.com/bailuk/java-gtk/blob/stage/java-gtk/src/main/resources/jna/loader.properties

It's also possible to add alternative library names during runtime: https://javadoc.jitpack.io/com/github/bailuk/java-gtk/0.5.0/javadoc/ch/bailu/gtk/lib/jna/Loader.html

It is described here in Section "Trouble shoot" / "Alternative names":
https://github.com/bailuk/java-gtk/blob/stage/doc/windows.md

Of course its not windows specific.
